pkg/controller/controller.go

```diff
@@ -186,20 +196,22 @@ func (c *Controller) clusterExists(name string, cl cloudprovider.Clusters) (bool
 
 // CreateComputePool create a compute node pool.
 func (c *Controller) CreateComputePool(p model.ComputePool) error {
-	cl, impl := c.Cloud.Clusters()
-	if !impl {
-		return ErrNotImplemented
-	}
 	pooler, impl := c.Cloud.NodePooler()
 	if !impl {
 		return ErrNotImplemented
 	}
 
-	// TODO Check if a masterpool for given cluster exists first? Maybe we
-	// don't care if masterpool does not exist yet, as long as the cluster infra exists?
-	if exists, err := c.clusterExists(p.ClusterName, cl); !exists {
+	clusters, err := c.GetClusters(p.ClusterName)
```

Comment:
Maybe a single method c.GetCluster wrapping GetClusters or rename GetClusters (as it's redundant as there's no wildcard option)...